src/test/regress/expected/citus_local_tables_queries.out

```diff
@@ -0,0 +1,1008 @@
+BEGIN;
+	TRUNCATE distributed_table;
+	-- should error out as we truncated distributed_table via parallel execution
+	TRUNCATE citus_local_table  CASCADE;
+NOTICE:  truncate cascades to table "reference_table"
+NOTICE:  truncate cascades to table "distributed_table"
+NOTICE:  executing the command locally: TRUNCATE TABLE citus_local_table_queries.citus_local_table_xxxxx CASCADE
+NOTICE:  truncate cascades to table "reference_table_xxxxx"
+NOTICE:  executing the command locally: TRUNCATE TABLE citus_local_table_queries.reference_table_xxxxx CASCADE
+ERROR:  cannot execute DDL on table "reference_table" because there was a parallel DDL access to distributed table "distributed_table" in the same transaction
+ROLLBACK;
+BEGIN;
+	SET LOCAL citus.multi_shard_modify_mode TO 'sequential';
+	TRUNCATE distributed_table;
+	-- should work fine as we already switched to sequential execution
+	-- before parallel truncate
+	TRUNCATE citus_local_table  CASCADE;
+NOTICE:  truncate cascades to table "reference_table"
+NOTICE:  truncate cascades to table "distributed_table"
+NOTICE:  executing the command locally: TRUNCATE TABLE citus_local_table_queries.citus_local_table_xxxxx CASCADE
+NOTICE:  truncate cascades to table "reference_table_xxxxx"
+NOTICE:  executing the command locally: TRUNCATE TABLE citus_local_table_queries.reference_table_xxxxx CASCADE
+ROLLBACK;
+ALTER TABLE distributed_table DROP CONSTRAINT fkey_dist_to_ref;
+BEGIN;
+    INSERT INTO citus_local_table VALUES (1) ON CONFLICT (a) DO NOTHING;
+NOTICE:  executing the command locally: INSERT INTO citus_local_table_queries.citus_local_table_1509000 AS citus_table_alias (a) VALUES (1) ON CONFLICT(a) DO NOTHING
+	-- FIXME: it would show "sequential" due to the bug:
+	-- https://github.com/citusdata/citus/issues/4138
+	-- normally, we should see "parallel" as no distributed
+	-- tables are involved in foreign key chain (transitively)
+    show citus.multi_shard_modify_mode;
+ citus.multi_shard_modify_mode
+---------------------------------------------------------------------
+ sequential
+(1 row)
+
+ROLLBACK;
+-- remove uniqueness constraint and dependent foreign key constraint for next tests
+ALTER TABLE reference_table DROP CONSTRAINT fkey_ref_to_local;
+NOTICE:  executing the command locally: SELECT worker_apply_inter_shard_ddl_command (1509002, 'citus_local_table_queries', 1509000, 'citus_local_table_queries', 'ALTER TABLE reference_table DROP CONSTRAINT fkey_ref_to_local;')
+ALTER TABLE citus_local_table DROP CONSTRAINT pkey_c;
+NOTICE:  executing the command locally: SELECT worker_apply_shard_ddl_command (1509000, 'citus_local_table_queries', 'ALTER TABLE citus_local_table DROP CONSTRAINT pkey_c;')
+COPY  citus_local_table(a) FROM  PROGRAM 'seq 32';
+-- should use local execution
+BEGIN;
+	COPY  citus_local_table(a) FROM  PROGRAM 'seq 32';
+NOTICE:  executing the copy locally for shard xxxxx
+	COPY  citus_local_table(a) FROM  PROGRAM 'seq 32';
+NOTICE:  executing the copy locally for shard xxxxx
```

Comment:
few missing tests:
- create index
- create constraint (like check constraint) / validate constraint
- INSERT INTO citus_local SELECT * FROM citus_local LIMIT 5;,
- function call delegation -- that we have implemented for reference tables, should that work with Citus local tables?
- Multi-row INSERTs: INSERT INTO local  VALUES (1,2),(3,4),(5,6);
- VACUUM
-  simple savepoint examples, which involves distributed/reference/citus local tables.

If we already have tests for any of the above, feel free to skip